### PR TITLE
POSTGRESQL AND LAYERED ARCHITECTURE

### PR DESCRIPTION
**Pull request by Nadya1993**
- Used a free web hosting services for PostgreSQL https://www.elephantsql.com/plans.html.
connectionString = 'postgres://bqgqthbd:9gOdo8uMOO6yQFufa-ypy2bFDW3SgeXz@dumbo.db.elephantsql.com:5432/bqgqthbd'

- Wrote SQL script which creates Users table in the DB and fill it in with predefined users’collection.
- Configured REST service to work with PostgreSQL.
- Used the sequelize package(http://docs.sequelizejs.com/)as ORM to work with PostgreSQL.
- The service adheres to 3-layer architecture principles (https://softwareontheroad.com/ideal-nodejs-project-structure/)

Postman collection: https://www.getpostman.com/collections/6211e69d33fdedede493

### CODE REVIEW

File: src/routers/suggestionRoutes.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import express from 'express';
+import { UserModel } from '../models/UserModel';
+import { UserService } from '../services/UserService';
+
+const router = express.Router();
+const UserServiceInstance = new UserService(UserModel);
```
Comment (spiderpoul) [inline src/routers/suggestionRoutes.ts]: seems like you have 2 equal instances of `UserService`, I suggest creating a singleton of `UserService` or keep `userRoutes`  and `suggestionRoutes` in the same file.

File: src/models/UserModel.ts [hunk 2]
```diff
@@ -1,99 +1,27 @@
...
-      login: String(login),
-      password: String(password),
-      age: Number(age),
-      isDeleted: false
-    },
-    ...this.users]
+import { Sequelize, DataTypes } from 'sequelize';
+import { UserInstance } from '../types';
+
+const connectionString = 'postgres://bqgqthbd:9gOdo8uMOO6yQFufa-ypy2bFDW3SgeXz@dumbo.db.elephantsql.com:5432/bqgqthbd';
```
Comment (spiderpoul) [inline src/models/UserModel.ts]: for future suggest to keep those private settings in non-tracked `.env` file. `.env-sample` file should be provided with example configs in this case. Checkout `dotenv ` https://www.npmjs.com/package/dotenv.
This is optional.

File: src/data-access/properties.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { Client } from 'pg';
+import { Sequelize } from 'sequelize';
+
+const connectionString = 'postgres://bqgqthbd:9gOdo8uMOO6yQFufa-ypy2bFDW3SgeXz@dumbo.db.elephantsql.com:5432/bqgqthbd';
+
+export const sequelize = new Sequelize(connectionString);
```
Comment (spiderpoul) [inline src/data-access/properties.ts:6]: suggesting creating singleton of this and export `sequelize`  only, seems like you create multiple instances of same connection

File: src/routers/suggestionRoutes.ts (hunk 1 above)
Comment (Nadya1993) [reply src/routers/suggestionRoutes.ts]: Fixed

File: src/models/UserModel.ts (hunk 2 above)
Comment (Nadya1993) [reply src/models/UserModel.ts]: Used .env file for private settings

File: src/data-access/properties.ts (hunk 3 above)
Comment (Nadya1993) [reply src/data-access/properties.ts:6]: Fixed

**Pull request merged**